From 9f98ab351951fb8fe81ee3771b17697f84f36347 Mon Sep 17 00:00:00 2001
Message-Id: <9f98ab351951fb8fe81ee3771b17697f84f36347.1441918786.git.yunwei.zhang@intel.com>
In-Reply-To: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
References: <8631704be06e00cb7a7a415f13811ca4ece99e8b.1441918786.git.yunwei.zhang@intel.com>
From: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
Date: Tue, 8 Sep 2015 12:17:39 +0530
Subject: [PATCH 07/10] REVERTME [VPG]: drm/i915: Disabling interrupts on
 disable call

This patch removes the enabling of interrupts done while disable
interrupts is called.

REVERTME: LP audio is not available upstream and may not be for
some more time which is required for this.

Change-Id: I91f864666f456ac5d5f9b436e15562033c2d8c30
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-10172
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Signed-off-by: Shubhangi Shrivastava <shubhangi.shrivastava@intel.com>
---
 drivers/gpu/drm/i915/hdmi_audio_if.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/hdmi_audio_if.c b/drivers/gpu/drm/i915/hdmi_audio_if.c
index d054951..6a4a5ed 100644
--- a/drivers/gpu/drm/i915/hdmi_audio_if.c
+++ b/drivers/gpu/drm/i915/hdmi_audio_if.c
@@ -365,10 +365,7 @@ static int hdmi_audio_set_caps(enum had_caps_list set_element,
 			int_masks |= I915_HDMI_AUDIO_UNDERRUN_ENABLE;
 		dev_priv->hdmi_audio_interrupt_mask &= ~int_masks;
 
-		if (dev_priv->hdmi_audio_interrupt_mask)
-			i915_enable_hdmi_audio_int(dev, pipe);
-		else
-			i915_disable_hdmi_audio_int(dev, pipe);
+		i915_disable_hdmi_audio_int(dev, pipe);
 		break;
 	default:
 		break;
-- 
1.9.1

